source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +168,126 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+std::string TextBlock::ParseISO8601UsingRegex() const
+{
+	std::regex pattern("\\{\\{(DATE|TIME)\\((\\d{4})-{1}(\\d{2})-{1}(\\d{2})T(\\d{2}):{1}(\\d{2}):{1}(\\d{2})(Z|([+-])(\\d{2}):{1}(\\d{2}))(\\)|, S[Hh][Oo][Rr][Tt]\\)|, L[Oo][Nn][Gg]\\))\\}\\}");
```

Comment:
> L[Oo][Nn][Gg] [](start = 165, length = 14)

I think everything should be case insensitive. This pattern wouldn't work for date(..., sHoRt). would it? A potential solution to not make the regex any more complicated is to transform the text to lowercase, and replace the regex in the original text.

Reply:
good point

---
In reply to: [148859877](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148859877) [](ancestors = 148859877)